### PR TITLE
Fixed removeKeyInObject.js which has been breaking moment objects

### PR DESCRIPTION
**Pull request by shtelzerartem**
### What does it do?

Added if clause, which doesn't apply ```removeKeyInObject``` on moment objects. 

### Why is it needed?

Fixes ```removeKeyInObject``` function which has been breaking moment objects and didn't allow to save objects with datetime fields.

### Related issue(s)/PR(s)

- #9032 
- #9116

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/utils/removeKeyInObject.js [hunk 1]
```diff
@@ -20,7 +20,11 @@ const removeKeyInObject = (obj, keyToRemove) => {
       return { ...acc, [current]: value.map(obj => removeKeyInObject(obj, keyToRemove)) };
     }
 
-    if (typeof acc[current] === 'object') {
+    if (typeof value === 'object') {
+      if (value._isAMomentObject === true) {
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/utils/removeKeyInObject.js:24]: Can you add a test for the following statement?

Thanks a lot for your contribution

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/strapi-plugin-content-manager/admin/src/utils/removeKeyInObject.js (hunk 1 above)
Comment (shtelzerartem) [reply packages/strapi-plugin-content-manager/admin/src/utils/removeKeyInObject.js:24]: For sure, thank you a lot, bro!

Comment (shtelzerartem) [reply packages/strapi-plugin-content-manager/admin/src/utils/removeKeyInObject.js:24]: Hey, I have added test for this specific case.

**Review by HichamELBSI — APPROVED** [review]
LGTM, thank you for your contribution

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**